src/pkg/services/m365/api/teamsChats_pager.go

```diff
@@ -0,0 +1,172 @@
+package api
+
+import (
+	"context"
+
+	"github.com/alcionai/clues"
+	"github.com/microsoftgraph/msgraph-sdk-go/chats"
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+	"github.com/microsoftgraph/msgraph-sdk-go/users"
+
+	"github.com/alcionai/corso/src/pkg/services/m365/api/graph"
+	"github.com/alcionai/corso/src/pkg/services/m365/api/pagers"
+)
+
+// ---------------------------------------------------------------------------
+// chat message pager
+// ---------------------------------------------------------------------------
+
+// delta queries are not supported
+var _ pagers.NonDeltaHandler[models.ChatMessageable] = &chatMessagePageCtrl{}
+
+type chatMessagePageCtrl struct {
+	chatID  string
+	gs      graph.Servicer
+	builder *chats.ItemMessagesRequestBuilder
+	options *chats.ItemMessagesRequestBuilderGetRequestConfiguration
+}
+
+func (p *chatMessagePageCtrl) SetNextLink(nextLink string) {
+	p.builder = chats.NewItemMessagesRequestBuilder(nextLink, p.gs.Adapter())
+}
+
+func (p *chatMessagePageCtrl) GetPage(
+	ctx context.Context,
+) (pagers.NextLinkValuer[models.ChatMessageable], error) {
+	resp, err := p.builder.Get(ctx, p.options)
+	return resp, graph.Stack(ctx, err).OrNil()
+}
+
+func (p *chatMessagePageCtrl) ValidModTimes() bool {
+	return true
+}
+
+func (c Chats) NewChatMessagePager(
+	chatID string,
+	cc CallConfig,
+) *chatMessagePageCtrl {
+	builder := c.Stable.
+		Client().
+		Chats().
+		ByChatId(chatID).
+		Messages()
+
+	options := &chats.ItemMessagesRequestBuilderGetRequestConfiguration{
+		QueryParameters: &chats.ItemMessagesRequestBuilderGetQueryParameters{},
+		Headers:         newPreferHeaders(preferPageSize(maxNonDeltaPageSize)),
+	}
+
+	if len(cc.Select) > 0 {
+		options.QueryParameters.Select = cc.Select
+	}
+
+	if len(cc.Expand) > 0 {
+		options.QueryParameters.Expand = cc.Expand
+	}
+
+	return &chatMessagePageCtrl{
+		chatID:  chatID,
+		builder: builder,
+		gs:      c.Stable,
+		options: options,
+	}
+}
+
+// GetChatMessages fetches a delta of all messages in the chat.
+func (c Chats) GetChatMessages(
+	ctx context.Context,
+	chatID string,
+	cc CallConfig,
+) ([]models.ChatMessageable, error) {
+	ctx = clues.Add(ctx, "chat_id", chatID)
+	pager := c.NewChatMessagePager(chatID, cc)
+	items, err := pagers.BatchEnumerateItems[models.ChatMessageable](ctx, pager)
+
+	return items, graph.Stack(ctx, err).OrNil()
+}
+
+// GetChatMessageIDs fetches a delta of all messages in the chat.
+// returns two maps: addedItems, deletedItems
+func (c Chats) GetChatMessageIDs(
+	ctx context.Context,
+	chatID string,
+	cc CallConfig,
+) (pagers.AddedAndRemoved, error) {
+	aar, err := pagers.GetAddedAndRemovedItemIDs[models.ChatMessageable](
```

Comment:
For my own understandiing, do chats return deleted items?

Reply:
Correct.  Deleted items have a non-zero value for `deletedDateTime` and no content, but are otherwise available in the list of messages.